modules/plugins/languages/clojure.nix

```
@@ -0,0 +1,56 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.meta) getExe;
+  inherit (lib.lists) isList;
+  inherit (lib.types) either listOf package str;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.lua) expToLua;
+
+  cfg = config.vim.languages.clojure;
+in {
+  options.vim.languages.clojure = {
+    enable = mkEnableOption "Clojure language support";
+
+    treesitter = {
+      enable = mkEnableOption "Clojure treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "clojure";
+    };
+
+    lsp = {
+      enable = mkEnableOption "Clojure LSP support" // {default = config.vim.lsp.enable;};
+      package = mkOption {
+        description = "Clojure LSP";
+        type = either package (listOf str);
+        default = pkgs.clojure-lsp;
```

Super minor nit, order should be:
- type
- default
- description
- example (if present)

```suggestion
        type = either package (listOf str);
        default = pkgs.clojure-lsp;
        description = "Clojure LSP";
```

otherwise, looks good to me! :)